### PR TITLE
feat: [M3-8016] - Add TagSelect to edit images drawer

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
## Description 📝
Adds a TagSelect to the edit image drawer based on the new field in the API.

## Changes  🔄
- Add TagSelect to edit image drawer
- Update APIv4 Image type and factory with new `tags` field
- Clean up `ImageDrawer` component, removing unused code

## Target release date 🗓️
5/28

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-05-14 at 1 52 31 PM | Screenshot 2024-05-14 at 1 52 19 PM |

### Prerequisites
- Configure Cloud Manager to use the alpha API (e.g., via DevTools)

### Verification steps
- Navigate to the Images page
- Click on the Action menu for an image and select 'Edit'
- Verify tag select appears
- Add/remove tags and click "Save Changes"
- Re-open edit drawer and verify changes are reflected
- Verify that there are no regressions in the Restore from Image drawer, which reuses this component

## TODO
- [x] Create unit tests for ImageDrawer

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/TagsInput/TagsInput.tsx [hunk 1]
```diff
@@ -42,15 +42,15 @@ export interface TagsInputProps {
   /**
    * Callback fired when the value changes.
    */
-  onChange: (selected: Item[]) => void;
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/TagsInput/TagsInput.tsx:45]: By default, this type is `string | number`

File: packages/manager/src/features/Images/ImagesDrawer.tsx
Comment (hkhalil-akamai) [inline packages/manager/src/features/Images/ImagesDrawer.tsx:1]: The 'create' and 'imagize' modes of this drawer are no longer used, allowing for significant clean-up.

**Review by bnussman-akamai — COMMENTED** [review]
Tags are working well ✅ 

- Found a few things 🐛 
  - The `Cancel` button does not work for me currently
  - The loading state persists when the drawer is reopened
  
 

https://github.com/linode/manager/assets/115251059/3509a6eb-47d4-4231-a26a-f5f14a5ed41f

<hr />

We decided as a team that going forward that we'd like to keep drawers purpose very concise. So in this case, we could split up `ImageDrawer` into `RestoreImageDrawer` and `EditImageDrawer`. We **don't** need to do this now, but you can if you'd like to!  It could also give us a chance to rewrite these forms to use `react-hook-form`, but again, not required

**Comment by hkhalil-akamai** [comment]
>   * The `Cancel` button does not work for me currently
>   * The loading state persists when the drawer is reopened

Good catch, fixed! ✅

**Comment by hkhalil-akamai** [comment]
> We decided as a team that going forward that we'd like to keep drawers purpose very concise. So in this case, we could split up `ImageDrawer` into `RestoreImageDrawer` and `EditImageDrawer`. We **don't** need to do this now, but you can if you'd like to! It could also give us a chance to rewrite these forms to use `react-hook-form`, but again, not required

There is definitely room for cleanup in this component! Making this PR as WIP while I address this.

**Review by bnussman-akamai — APPROVED** [review]
Tagging looks good! 🏷️ 🎉 

I did notice this behavior shown in the video. Clearing a field and saving results in no change to the Image. I'd expect a validation error if I clear the label field. We can address this later

https://github.com/linode/manager/assets/115251059/25580481-c277-456a-946d-f83cce006f06

**Review by abailly-akamai — APPROVED** [review]
Great PR and cleanup!

left a couple code comments.

Also, to add to either this PR or the follow up ticket you created, when encountering an API error (ex: block PUT | https://api.dev.linode.com/v4/images/private requests), the error will persist when reopening the drawer.

Screen Shot 2024-05-21 at 15 58 54

Lastly, it's pretty strange to be able to add to add/edit tags in create/edit flow yet they are not visible anywhere in the UI when looking at the image list (prolly a follow up with UX would be good there)

Screen Shot 2024-05-21 at 16 01 22

**Pull request merged**